index.js

```diff
@@ -68,8 +68,9 @@ function _createXHR(options) {
 
         if (xhr.response) {
             body = xhr.response
-        } else if (xhr.responseType === "text" || !xhr.responseType) {
-            body = xhr.responseText || xhr.responseXML
+        } else {
+            var xml = getXml(xhr)
+            body = xhr.responseText || xml
```

Comment:
if xhr.responseText is available, all work done by getXml is for nothing. 
If you roll it back to a single line, the function will only be called if needed.